CMakeLists.txt

```diff
@@ -1,11 +1,7 @@
 # CMake build script for ZeroMQ
-project(ZeroMQ)
+cmake_minimum_required(VERSION 2.8.12)
```

Comment:
the conditional also is needed otherwise osx breaks

Reply:
that doesn't work, as it breaks osx, and breaking backward compatibility doesn't work either

Reply:
According to the CMake maintainer (https://gitlab.kitware.com/cmake/cmake/-/issues/22671#note_1020156) the current state is broken as well.

Reply:
That's self-evidently not true, given it does work just fine

Reply:
So you are saying that it's not possible to support all 3 of these:

- keep backward compatibility
- keep OSX working
- add support for setting policies on the command line

That still sounds like a cmake bug/limitation to me. But anyway, the first two are really not negotiable, so you'll need to find a different fix if you want the 3rd one to work I'm afraid.

Reply:
In a way yes, if upping the minimum CMake version to the higher common value (3.0.8) is not acceptable. I am not however that intimate with CMake to claim what exactly does not work when the things are called in the wrong order. Setting the policies on the command line not working could be just one of the consequences. Anyway, your call.